client/src/components/FoodSeeker/SearchResults/AdvancedFilters/AdvancedFilterButton.js

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Button } from "@mui/material";
+import { styled } from '@mui/material/styles'
+
+const DropdownIcon = () => (
+  <svg xmlns="http://www.w3.org/2000/svg" width="10" height="7" viewBox="0 0 10 7" fill="none">
+    <path d="M1 1.75L3.50518 4.56833C4.30076 5.46336 5.69924 5.46336 6.49482 4.56833L9 1.75" stroke="#1B1B1B" stroke-width="2" stroke-linecap="round" />
+  </svg>
+);
+const CustomButton = styled(Button)(({ isSelected, theme }) => ({
+  "&.MuiButton-contained": {
+    fontSize: "0.875rem",
+    borderRadius: "1.875rem",
+    letterSpacing: "0.035rem",
+    textTransform: "none",
+    gap: "0.5rem",
+    height: "2rem",
+    boxShadow: "0px 4px 10px 0px rgba(0, 0, 0, 0.20)",
+    border: isSelected ?
+      theme.palette.filterButtons.border.selected : theme.palette.filterButtons.border.notSelected,
+    color: theme.palette.filterButtons.bodyText,
+    backgroundColor: isSelected ?
+      theme.palette.filterButtons.backgroundColor :
+      theme.palette.common.white,
+    "&:hover": {
+      backgroundColor:
+        theme.palette.filterButtons.backgroundColor,
+    }
+  }
+}));
+
+const AdvancedFilterButton = ({
+  label,
+  onClick,
+  icon,
+  hasDropdown,
+  isSelected,
+  disabled,
+}) => {
+
+  const Icon = icon;
+
+  return (
+    <CustomButton
+      variant="contained"
+      disabled={disabled}
+      isSelected={isSelected}
+      onClick={(event) => onClick(event)}
+    >
+      {icon !== "" && <Icon />}
```

Comment:
I think this can just be `icon && <Icon />`. This way we don't need to pass in `icon=""` as a prop. It'll just be an optional prop

Reply:
I made all of the recommended changes **except:** 

- the 1st issue you brought up (rendering a view with a selected organization before shows the list). _I think that there is an organization "hidden" behind the button. I will investigate and find a solution._

- For your comment "button also keeps the selected style even after deselecting", _I will investigate this too_.

Reply:
For problem where marker underneath the filter buttons was being clicked, I removed the filter buttons from 'inside' the map and moved it to a absolute position 'next' to the map.

I can't reproduce the "button also keeps the selected style even after deselecting".